src/rules.c

```diff
@@ -451,11 +459,11 @@ void SchemaRule_RdbSave(SchemaRule *rule, RedisModuleIO *rdb) {
   RedisModule_SaveStringBuffer(rdb, ruleTypeStr, strlen(ruleTypeStr) + 1);
   RedisModule_SaveUnsigned(rdb, array_len(rule->prefixes));
   for (size_t i = 0; i < array_len(rule->prefixes); ++i) {
-    RedisModule_SaveStringBuffer(rdb, rule->prefixes[i], sdslen(rule->prefixes[i]) + 1);
+    HiddenUnicodeString_SaveToRdb(rule->prefixes[i], rdb);
```

Comment:
I prefer to avoid this trait at hidden strings
you can always do get_raw and get the underlying data, same as normal hidden string.